### PR TITLE
Posthog Analytics

### PR DESCRIPTION
**Pull request by PaulMcInnis**
This migrates from MixPanel analytics over to PostHog, and changes some events to include more data per the [spec](https://www.notion.so/berachain/PostHog-docs-1276120abced805aa7cfe8f0753a13cd). 

ex:

`swap_BERA` event
https://eu.posthog.com/project/40908/events/0193741e-4563-7bf1-bedf-c1709ae7090b/2024-11-28T18%3A52%3A41.342000Z

image

edit: this also dumps mixpanel analytics, including their packages, but since react 19 has released and we're not trying to upgrade that right now, i've pinned those dependencies to <19 (we can [upgrade](https://berachain-fe.atlassian.net/browse/BFE-404) later)

### CODE REVIEW

**Comment by PaulMcInnis** [comment]
OK, this is effectively `blocked` until we decide if we want to change the events to match the new incoming ones

**Comment by PaulMcInnis** [comment]
OK this first PR will focus on a subset of events 👍

File: apps/hub/src/components/swap-card.tsx [hunk 1]
```diff
@@ -204,10 +204,18 @@ export function SwapCard({
...
-      track("swap_token_success", {
-        tokenFrom: selectedFrom?.symbol,
-        tokenTo: selectedTo?.symbol,
-      });
+      if (!selectedFrom?.symbol || !selectedTo?.symbol) {
+        captureException(new Error("swap_token_with_unknown_symbols"));
+      } else if (!fromAmount || !toAmount) {
+        captureException(new Error("swap_token_with_unknown_amounts"));
+      } else {
+        track(`swap_${selectedFrom.symbol}`, {
```
Comment (PaulMcInnis) [inline apps/hub/src/components/swap-card.tsx:212]: This should instead be swap_token, and we will keep the payload

**Comment by PaulMcInnis** [comment]
This adds support for the following events:

- Swap
- Redeem (swap page)
- add/withdraw from Pool
- Stake/Unstake from Vault section (worded as deposit/withdraw currently)

**Comment by Bearlinguer** [comment]
Cannot see Deposit event, only the clicks around that. The flow is right but don't have "deposit" as a event

File: packages/shared-ui/src/utils/analytics.ts [hunk 2]
```diff
@@ -28,6 +33,28 @@ if (isMixpanelEnabled) {
...
+// Initialize PostHog
+const isPosthogEnabled = postHogProjectKey && !isDevelopmentWithoutAnalytics;
+
+if (isPosthogEnabled) {
+  posthog.init(postHogProjectKey, {
+    api_host: postHogHostAddress,
+    debug: true,
+    persistence: "localStorage",
+  });
+  console.log(
```
Comment (PaulMcInnis) [inline packages/shared-ui/src/utils/analytics.ts:45]: dump this

File: apps/hub/src/app/pools/[poolId]/withdraw/WithdrawPageContent.tsx [hunk 3]
```diff
@@ -192,7 +219,7 @@ export default function WithdrawLiquidityContent({
   return (
     <div className="mt-16 flex w-full flex-col items-center justify-center gap-4">
       {ModalPortal}
-      <Card className="mx-6 w-full items-center bg-background p-4 sm:mx-0 sm:w-[480px] flex flex-col">
+      <Card className="mx-6 flex w-full flex-col items-center bg-background p-4 sm:mx-0 sm:w-[480px]">
```
Comment (PaulMcInnis) [inline apps/hub/src/app/pools/[poolId]/withdraw/WithdrawPageContent.tsx:222]: alot of these changes are formatting which really ought to be applied project wide.

**Review by PaulMcInnis — COMMENTED** [review]
👀

**Review by bearpong — CHANGES_REQUESTED** [review]
gmgm, can we remove mixpanel in this PR?

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -6,6 +6,9 @@ settings:
 
 overrides:
   rpc-websockets: ^7.11.1
+  react: <19
```
Comment (PaulMcInnis) [inline pnpm-lock.yaml:9]: we should upgrade to version 19 ([task](https://berachain-fe.atlassian.net/browse/BFE-404), [guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide))

**Review by PaulMcInnis — COMMENTED** [review]
👀

**Comment by PaulMcInnis** [comment]
phew! OK, sorted the issue with static build, it seems that we needed to be using node 20 in that GH action. 

RE the node 19/18 debacle I've cut a separate PR to address that #255

**Review by bearpong — APPROVED** [review]

**Pull request merged**